v22.1/vault-db-secrets-tutorial.md

```diff
@@ -0,0 +1,441 @@
+Set your CRDB cluster credentials and other configuration information as environment variables
+{% include_cached copy-clipboard.html %}
+```shell
+USER_NAME= # CRDB admin username
+PASSWORD= # CRDB admin password
+DB_NAME=defaultdb
+CLUSTER_NAME=lilac-grizzly-684
+HOST=free-tier123.aws-us-west-2.crdb.io
+
+```
+#### Obtain your CRDB cluster's CA public certificate
+
+1. Visit the [CockroachDB Cloud Console's cluster page.](https://cockroachlabs.cloud/cluster/). 
+1. Select your cluster.
+1. Click the **Connect** button.
+1. Select **"Download CA Cert (Required only once)"** and use the generated `curl` command to download the certificate.
+```shell
+curl --create-dirs -o root.crt -O https://management-staging.crdb.io/clusters/505a138c-37ff-46b7-9c50-4119cf0881f6/cert
+```
+1. Place the certificate on the path specified by `sslrootcert` in the following command, the run it. A successful display of the output of "show tables;" proves that our connection works.
+
+{% include_cached copy-clipboard.html %}
+```shell
+TLS_OPTS="sslrootcert=root.crt&sslmode=verify-full&options=--cluster=${CLUSTER_NAME}"
+export CLI_DB_CONNECTION_URL="postgresql://$USER_NAME:$PASSWORD@${HOST}:26257/${DB_NAME}?${TLS_OPTS}"
+cockroach sql --url "${CLI_DB_CONNECTION_URL}" --execute "show tables;"
+```
+
+```txt
+SHOW TABLES 0
+
+Time: 107ms
+```
+
+
+### Connect to Vault
+
+Set your Vault target and authenticate with an admin token.
+You can fetch your target URL and generate a token from the [Hashicorp Vault console](https://portal.cloud.hashicorp.com/services/vault).
+
+
+{% include_cached copy-clipboard.html %}
+```shell
+export VAULT_ADDR=https://roach-test-vault.vault.bfb2290a-670b-4a10-bedf-5aab18e84d69.aws.hashicorp.cloud:8200 # your Vault cluster URL
+export VAULT_NAMESPACE=admin
+vault login
+```
+```shell
+Success! You are now authenticated. The token information displayed below
+is already stored in the token helper. You do NOT need to run "vault login"
+again. Future Vault requests will automatically use this token.
+```
+
+### Enable the Vault database secrets engine
+
+This only needs to be done once for an individual Vault cluster.
+
+{% include_cached copy-clipboard.html %}
+```shell
+vault secrets enable database
+```
+
+```txt
+Success! Enabled the database secrets engine at: database/
+```
+
+## Binding your Vault cluster to your CRDB cluster
+
+The connection lies in a Vault configuration, which will store your CRDB admin credentials, allowing Vault to administrate CRDB credentials.
+
+### Create a Vault database configuration
+
+Create the `crdb-config` database configuration in Vault, specifying admin credentials that will be used by Vault to create credentials for your defined role.
+
+{% include_cached copy-clipboard.html %}
+```shell
+TLS_OPTS="sslmode=require&options=--cluster=${CLUSTER_NAME}"
+export VAULT_DB_CONNECTION_URL="postgresql://{{username}}:{{password}}@${HOST}:26257/${DB_NAME}?${TLS_OPTS}"
+
+vault write database/config/crdb-config \
```

Comment:
Currently, here, I'm running into a `400` error trying to verify the connection. As a side, everything was good until this point, including executing `cockroach sql`. Thought I would let you take a look through my notes until this point and I'll run through it again tomorrow, or after any changes you want to make.